### ISSUE TITLE
TeamTalk Server is unstable on AMD EPYC processors

### ISSUE THREAD

**Issue by tech10** [BearWare/TeamTalk5#808]
I have consistently seen results that the TeamTalk server isn't completely stable on AMD EPYC processors, I don't believe the server version is particularly relevant to this issue. This can easily be tested by others using a Linode if desired, as a great deal of them are now using AMD processors on their hosts. I have observed the following issues.

- Intermittent jitter or latency increase on connected clients, anywhere between a few minutes to a few hours after the clients are connected.
- Connection drops intermittently, though this one is less common.

I hope these can be tested and reproduced, and if possible, corrected. Thank you.

**Comment by bear101** [comment]
"Server"? Is this the hardware server or software server? If the server software crashes then it's something we can fix. If the hardware server is unstable then there's nothing we can do.

**Comment by tech10** [comment]
This is the TeamTalk server itself. The hardware is fine. I've changed the title to clarify that a bit.

**Comment by adns44** [comment]
I host my own TT5 server on my VPS from Contabo and on another VPS for my friend. I have AMD Epyc CPU, my friend has Intel Xeon. We haven't any problem. My server used by 10--15 users simultaneous, and we don't have problem. Some users are logged in for few days.

**Comment by tech10** [comment]
I know I have had the difficulties mentioned above. In fact, the day I opened this issue, I was running two servers on a single VPS with an AMD EPYC CPU. Both of them were dropping connections until I switched to Intel, where they have remained stable. I have also had the difficulty of occasional buffering or increased latency occurring with far greater frequency on a server running on an AMD EPYC, than that running on an Intel, and in that particular case, the server didn't drop connections, and most people either didn't notice or didn't care about the audio instability.

Results seem inconsistent.

**Comment by bear101** [comment]
@tech10 I very much doubt the TeamTalk server binary has anything to do with this. TeamTalk doesn't favour any CPU.

**Comment by tech10** [comment]
@bear101 While that may be so, I have consistently experienced instability of one degree or another, whether to a lesser or greater degree, while on AMD compared to servers running on Intel. These servers are running on Linode's, a VPS running under KVM. All other servers and network performance on the AMD servers perform perfectly, it's only the TeamTalk server that seems to have instabilities.

For instance, while my two TeamTalk servers were dropping connections, I was able to maintain a connection to my server over SSH, which is TCP, and over Mosh, which is UDP. The connections were stable without any extra latency. Pinging also worked perfectly. When I changed the server to host on an Intel CPU over AMD, the TeamTalk servers stabilized. At the moment, that's all I can state regarding the matter, but if there is any way I could perform a deeper analysis, I'll be glad to attempt it.
